src/openforms/middleware.py

```diff
@@ -18,3 +22,24 @@ def __call__(self, request):
             value = SAMESITE_VALUE if origin and not settings.DEBUG else "Lax"
             response.cookies[settings.SESSION_COOKIE_NAME]["samesite"] = value
         return response
+
+
+class SessionTimeoutMiddleware:
+    """
+    Allows us to set the expiry time of the session based on what
+    is configured in our GlobalConfiguration
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        config = GlobalConfiguration.get_solo()
+        timeout = (
+            config.admin_session_timeout
+            if request.user.is_staff
+            else config.form_session_timeout
+        )
+        # https://docs.djangoproject.com/en/2.2/topics/http/sessions/#django.contrib.sessions.backends.base.SessionBase.set_expiry
+        request.session.set_expiry(timedelta(minutes=timeout).seconds)
```

Comment:
I'll leave this here :-)

```
>>> from datetime import timedelta
>>> timedelta(weeks=1, seconds=5).seconds
5
>>> timedelta(weeks=1, seconds=5).total_seconds()
604805.0
```

Reply:
FYI - this doesn't affect us really because we pass `minutes` which is converted into seconds, however it's definitely more robust to use `total_seconds()` in case the call ever changes